internal/langserver/handlers/command/validate.go

```diff
@@ -0,0 +1,49 @@
+package command
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/creachadair/jrpc2/code"
+	lsctx "github.com/hashicorp/terraform-ls/internal/context"
+	"github.com/hashicorp/terraform-ls/internal/langserver/cmd"
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+)
+
+func TerraformValidateHandler(ctx context.Context, args cmd.CommandArgs) (interface{}, error) {
+	dirUri, ok := args.GetString("uri")
+	if !ok || dirUri == "" {
+		return nil, fmt.Errorf("%w: expected dir uri argument to be set", code.InvalidParams.Err())
+	}
+
+	dh := ilsp.FileHandlerFromDirURI(lsp.DocumentURI(dirUri))
+
+	cf, err := lsctx.RootModuleFinder(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	rm, err := cf.RootModuleByPath(dh.Dir())
+	if err != nil {
+		return nil, err
+	}
+
+	wasInit, _ := rm.WasInitialized()
```

Comment:
I remember we had a conversation about this, but what was the reasoning behind ignoring the error here? Would we not want to differentiate e.g. between a state where the module doesn't exist at all and when it exists, but isn't initialized?

Reply:
Is there really a difference? As long as the URI is a folder, it is either initialized and a place where you can call terraform validate, or it isn't?

Reply:
Well, in most cases there probably won't be a difference, but that's just because we rely on the client always calling `didOpen` on _a_ file from any folder that we run validate on.

In theory a client might integrate the command in such a way that it allows user to run it in folders which were never opened in the editor (but may in fact be initialized).

We should probably somehow account for this edge case and automatically load the module and make it known, but I think that raising a correct error is a solid start?

Reply:
Ahh okay makes sense now, I will raise the error